File: 04_quicksort/swift/03_recursive_count.swift
import Foundation

func count(list : [Int]) -> Int {
    if list.count == 0 {
        return 0
    }
    var tempArray = list
    tempArray.remove(at: 0)
    return 1 + count(list: tempArray)
}

print(count(list: [1, 2, 3, 4])) // => 4

File: 04_quicksort/golang/03_recursive_count.go
package main

import "fmt"

func count(list []int) int {
	if len(list) == 0 {
		return 0
	}
	return 1 + count(list[1:])
}

func main() {
	fmt.Println(count([]int{0, 1, 2, 3, 4, 5}))
}
